### ISSUE TITLE
Training Dataset Creation for Non-toy Rigs

### ISSUE THREAD

**Issue by fire** [electronicarts/rig-inversion#1]
## Problem

Write documentation for guidance on generating training datasets with complex, production-level rigs.

### Current Steps

```python
# Generate a toy dataset:
python generate_toy_dataset.py
# Train and test rig model:
python train_rig_approximation.py
# Inverse rig using the trained model:
python inverse_rig.py
```

These steps don't apply to non-toy rigs, which poses a problem for users who need to work with more advanced setups.

Edited:

> Rig inversion is a mathematical approach that allows animators to remap an existing mesh animation onto an animation rig. This allows animators to tweak and fix up existing mesh animations, which would be difficult or impossible otherwise.
>
> The difficulty with rig inversion is finding the rig parameter vector that best approximates a given input mesh.

From: https://www.ea.com/seed/news/seed-rig-inversion-differentiable-rig-function

**Comment by fire** [comment]
I am investigating using the VRM rig. https://github.com/V-Sekai/ANIM_test_assets

1. [Using a Differentiable Function for Rig Inversion](https://www.youtube.com/watch?v=sYCz9LGIkuI)
1. [ANIM_mocap_vsekai_08.tres](https://github.com/V-Sekai/ANIM_test_assets/blob/main/ANIM_mocap_vsekai_08.tres) is a skeleton animation file in Godot Engine animation format that can be exported to glb when overlayed on a rigged character with meshes.
2. [VRM1_vsekai_godot_engine_humanoid_08.vrm](https://github.com/V-Sekai/ANIM_test_assets/blob/main/VRM1_vsekai_godot_engine_humanoid_08.vrm) is a fake rig.
3. https://github.com/madjin/vrm-samples has a few sample VRM that have different meshes but can be using the same skeletal convention.
4. How can I use rig-inversion?

**Comment by fire** [comment]
As far as I understand the paper uses a vertex animation animation from motion capture of a face and using rig inversion find the parameters of an existing face rig.

**Comment by mathieumbea** [comment]
The steps remain the same regardless of the complexity of the rig. Please refer to the paper on how to best generate the rig training dataset. As a summary, you need to:
1. Know the set of rig input parameters and their respective bounds
2. Randomly sample the rig with those parameters within those bounds
3. Save the output of the rig along the input parameters

This implies that, though the rig remains a black box, you need to be able to procedurally execute it for selected inputs, through your animation software's scripting system for example, and record the output.

Its important to cover the entire input space of the rig, but our experiments suggest it is best to randomly sample the rig by activating a number of inputs following a power law. That is, most samples in the dataset should have few non-zero inputs, and few samples should have most inputs being non-zero.
